### PR TITLE
Adding more unit tests

### PR DESCRIPTION
**Pull request by Lomadriel**
- [x] Test with a key bigger than std::size_t
- [x] Multi thread tests

### CODE REVIEW

**Comment by Lomadriel** [comment]
Can't build until #26 is merged.

**Comment by Lomadriel** [comment]
Seems ready to merge.

File: include/wfc/unordered_map.hpp [hunk 1]
```diff
@@ -378,7 +376,7 @@ namespace wfc
 	auto unordered_map<Key, Value, HashFunction>::allocate_node(hash_t hash, key_t key, value_t value) const
 	    -> node_union
 	{
-		return node_union{new (std::align_val_t{8}) node_t{hash, key, value}};
+		return node_union{new /*(std::align_val_t{8})*/ node_t{hash, key, value}};
```
Comment (CBenoit) [inline include/wfc/unordered_map.hpp]: Why? Should we open an issue?

File: tests/unordered_map/multi_thread_test.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+		threads.emplace_back([this, i, &fails]() {
+			futur.wait();
+			std::size_t idx = i % 8;
+
+			for (std::size_t j = block_low(idx, nbr_threads / 2, map_size),
+			                 n = block_high(idx, nbr_threads / 2, map_size);
+			     j <= n;
+			     ++j)
+			{
+				if (map.update(static_cast<unsigned char>(j), 2 * (j + 1), j + 1) != wfc::operation_result::success)
```
Comment (CBenoit) [inline tests/unordered_map/multi_thread_test.cpp]: You can use `fail` helper. Same goes for other tests.

File: tests/unordered_map/multi_thread_test.cpp [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+			for (std::size_t j = block_low(idx, nbr_threads, map_size - 1),
+			                 n = block_high(idx, nbr_threads, map_size - 1);
+			     j <= n;
+			     ++j)
+			{
+				ASSERT_EQ(custom_map.insert(static_cast<unsigned char>(j), 2 * j), wfc::operation_result::success);
+			}
+
+			//printf("j: %zu, n: %zu\n", block_low(i, nbr_threads, map_size), block_high(i, nbr_threads, map_size));
```
Comment (CBenoit) [inline tests/unordered_map/multi_thread_test.cpp]: To remove.

File: tests/unordered_map/single_thread_test.cpp [hunk 4]
```diff
@@ -179,3 +179,52 @@ TEST(WaitFreeHashMap, FullHashMapVisist)
...
+	FatKeyHash& operator>>=(int shift) {
+		*this = *this >> shift;
+		return *this;
+	}
+
+	std::size_t operator&(std::size_t arg) {
+		return p2 & (arg);
+	}
+};
+bool operator==(const FatKeyHash& a, const FatKeyHash& b);
```
Comment (CBenoit) [inline tests/unordered_map/single_thread_test.cpp:211]: Check clangformat just to be sure.

**Review by CBenoit — CHANGES_REQUESTED** [review]

File: include/wfc/unordered_map.hpp (hunk 1 above)
Comment (Lomadriel) [reply include/wfc/unordered_map.hpp]: Corrected

**Review by CBenoit — APPROVED** [review]

**Pull request merged**